config/errors.go

```diff
@@ -91,3 +103,50 @@ type ErrInvalidURIPrefix string
 func (e ErrInvalidURIPrefix) Error() string {
 	return fmt.Sprintf("config: invalid uri_prefix (%v). uri_prefix must start with a forward slash '/' ", string(e))
 }
+
+// ErrUnknownProviderType is return when the config contains a provider type that has not been registered
+type ErrUnknownProviderType struct {
+	Name           string // name of the entry in the config
+	Type           string // name of the data provider
+	KnownProviders []string
+}
+
+func (e ErrUnknownProviderType) Error() string {
+	return fmt.Sprintf("config: invalid type (%s) for provider %s; known providers are: %v", e.Name, e.Type, strings.Join(e.KnownProviders, ","))
+}
+
+// Is returns weather the error is of type ErrUnknownProviderType, only checking the Type value.
+func (e ErrUnknownProviderType) Is(err error) bool {
+	err1, ok := err.(ErrUnknownProviderType)
+	if !ok {
+		return false
+	}
+	return err1.Type == e.Type
+}
```

Comment:
If the user is looking for a type comparison, shouldn't they use `As()`? It seems to me this is overriding `Is()` to make it function like `As()`

Reply:
It's not quite. As type as to be the same, it's not just that it's the same error type, but then the `Type` field has to be the same as well. 

`ErrUnknownProviderType{ Type: "foo" }` is different than `ErrUnknownProviderType{ Type: "bar" }`, the reason for the overload, is because the KnownProvider and Name fields are not important, and are only there to help with the error message.

Reply:
Yep, now that I re-read it I see what you're doing. e.Type is a property not the actual type of the struct.